client/verta/tests/test_deployment.py

```diff
@@ -321,16 +321,17 @@ def test_cached_fetch_zip(self, experiment_run, strs, dir_and_files):
     def test_fetch_tgz(self, experiment_run, strs, dir_and_files):
         dirpath, filepaths = dir_and_files
         key = strs[0]
-
-        with tempfile.NamedTemporaryFile(suffix='.tgz') as tempf:
-            # make archive
-            with tarfile.open(tempf.name, 'w:gz') as tarf:
-                tarf.add(dirpath, "")
-            tempf.flush()  # flush object buffer
-            os.fsync(tempf.fileno())  # flush OS buffer
-            tempf.seek(0)
-
-            experiment_run.log_artifact(key, tempf.name)
+        tempf  = tempfile.NamedTemporaryFile(suffix='.tgz', delete=False)
+        
+        # make archive
+        with tarfile.open(tempf.name, 'w:gz') as tarf:
+            tarf.add(dirpath, "")
+        tempf.flush()  # flush object buffer
+        os.fsync(tempf.fileno())  # flush OS buffer
+        tempf.seek(0)
+        tempf.close()
+
+        experiment_run.log_artifact(key, tempf.name)
```

Comment:
I suspect that for all of these, we could leave the `with:` block intact but simply move the `log_artifact() call out of it (unindent that one line, one level)